src/reducers/workspaces.ts

```diff
@@ -167,25 +251,24 @@ export const reducer: Reducer<IWorkspaceManagerState> = (
     case SEND_REPL_INPUT_TO_OUTPUT:
       // CodeOutput properties exist in parallel with workspaceLocation
       newOutput = state[location].output.concat(action.payload as CodeOutput)
-      return {
-        ...state,
-        [location]: {
-          ...state[location],
-          output: newOutput
-        }
+      let newReplHistoryRecords: string[]
+      if (action.payload.value !== '') {
+        newReplHistoryRecords = [action.payload.value].concat(state[location].replHistory.records)
+      } else {
+        newReplHistoryRecords = state[location].replHistory.records
       }
-    case EVAL_EDITOR:
-      return {
-        ...state,
-        [location]: {
-          ...state[location]
-        }
+      if (newReplHistoryRecords.length > 50) {
```

Comment:
You can consider using a deque or a circular buffer, so that the size does not have to be a concern.